Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/docrepository/adapter/service/DocumentServiceTest.java

```diff
@@ -168,55 +150,39 @@ public void testDocumentQuery() {
         when(EVENTCODE_DAO.eventCodeQuery(Mockito.anyList())).thenReturn(eventCodeList);
         when(eventcode.getDocument()).thenReturn(doc);
 
-        List<Document> resultDocList = documentService.documentQuery(dqParams);
-        Document resultDoc = resultDocList.get(0);
+        List<DocumentMetadata> resultDocList = documentService.documentQuery(dqParams);
+        DocumentMetadata resultDoc = resultDocList.get(0);
 
-        assertEquals(resultDocList.size(), 1);
-        assertEquals(resultDoc.getDocumentid(), (Long) DOC_ID);
-        assertEquals(resultDoc.getDocumentUniqueId(), DOC_UNIQUE_ID);
+        assertEquals(1, resultDocList.size());
+        assertEquals((Long) DOC_ID, resultDoc.getDocumentid());
+        assertEquals(DOC_UNIQUE_ID, resultDoc.getDocumentUniqueId());
     }
 
     @Test
     public void testGetAllDocuments() {
         DocumentService documentService = getDocumentServiceWithMockDaos();
 
-        List<Document> documents = new ArrayList<>();
+        List<DocumentMetadata> documents = new ArrayList<>();
 
         when(DOCUMENT_DAO.findAll()).thenReturn(documents);
 
-        List<Document> results = documentService.getAllDocuments();
+        List<DocumentMetadata> results = documentService.getAllDocuments();
 
-        assertTrue(results instanceof List);
+        verify(DOCUMENT_DAO).findAll();
+        assertEquals(documents, results);
     }
 
-    @Test
-    public void testGetDocumentDao() {
-        DocumentService documentService = new DocumentService();
-        DocumentDao documentDao = documentService.getDocumentDao();
-
-        assertTrue(documentDao instanceof DocumentDao);
-    }
+    private DocumentService getDocumentServiceWithMockDaos() {
 
-    @Test
-    public void testGetEventCodeDao() {
+        // Since we aren't using Autowiring for these types of things for some reason... Reflection to the rescue.
         DocumentService documentService = new DocumentService();
-        EventCodeDao eventCodeDao = documentService.getEventCodeDao();
-
-        assertTrue(eventCodeDao instanceof EventCodeDao);
-    }
+        Field docDao = ReflectionUtils.findField(DocumentService.class, "documentDao");
```

Comment:
Is there a reason why we're using reflection here instead of just overriding the get methods?

Reply:
Mostly personal preference. Overriding them in my opinion is too confusing and adds dozens of lines of code that could be  represented as to whats going on in the test much clearer with a correctly initialized mock. 
For example: https://github.com/CONNECT-Solution/CONNECT/pull/1642/files#diff-51ee2e33c2722f883d38bae4432e17d4L171

Over 300 lines got condensed into a very simple HashMap which is not only shorter, but much easier to understand.

Hopefully we will get to a point where all these dependencies are Autowired and we can simply annotate them with @Mock and call @InjectMocks on the class under the test.